pkg/virt-controller/watch/migration.go

```diff
@@ -1346,6 +1388,58 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 	return nil
 }
 
+func (c *MigrationController) addDeschedulerAnnotationToSourcePod(sourcePod *k8sv1.Pod) error {
+	if _, exists := sourcePod.GetAnnotations()[virtv1.DeschedulerEvictionInProgressAnnotation]; !exists {
+		patchSet := patch.New(
+			patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(virtv1.DeschedulerEvictionInProgressAnnotation)), ""),
+		)
+		patchBytes, err := patchSet.GeneratePayload()
+		if err != nil {
+			return err
+		}
+
+		_, err = c.clientset.CoreV1().Pods(sourcePod.Namespace).Patch(context.Background(), sourcePod.Name, types.JSONPatchType, patchBytes, v1.PatchOptions{})
+		if err != nil {
+			log.Log.Object(sourcePod).Errorf("failed to add %s pod annotation: %v", virtv1.DeschedulerEvictionInProgressAnnotation, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MigrationController) removeDeschedulerAnnotationFromSourcePod(migration *virtv1.VirtualMachineInstanceMigration) error {
+	podKey := controller.NamespacedKey(migration.Namespace, migration.Status.MigrationState.SourcePod)
+	obj, exists, err := c.podIndexer.GetByKey(podKey)
+	if !exists {
+		log.Log.Reason(err).Errorf("source pod %s does not exist", migration.Status.MigrationState.SourcePod)
+		return nil
+	}
+	if err != nil {
+		log.Log.Reason(err).Errorf("Failed to fetch source pod %s for namespace from cache.", migration.Status.MigrationState.SourcePod)
+		return nil
+	}
+
+	sourcePod := obj.(*k8sv1.Pod)
+	if _, exists := sourcePod.GetAnnotations()[virtv1.DeschedulerEvictionInProgressAnnotation]; exists {
+		patchSet := patch.New(
+			patch.WithTest(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(virtv1.DeschedulerEvictionInProgressAnnotation)), ""),
```

Comment:
i'm not sure that the `WithTest` is needed here

Reply:
It is better to check that we are dropping something that exists.

Reply:
You would use the `test` operation to make sure a condition is true before making changes. However, I’m not sure we need it here. Besides, we already checked that the annotation exists in the if condition. Could you please explain how the `test` operation improves this?

Reply:
The test operation is commonly used to prevent an update when there's a concurrency conflict.
Even if we should be the only one removing this annotation, it can happen that the resource has been modified during the controller loop, hence that we are working with an older version. In this case the `test` operation allows us to be consistent, doing the check server side too.

Reply:
> Even if we should be the only one removing this annotation, it can happen that the resource has been modified during the controller loop, hence that we are working with an older version

why is that a problem?

This topic is important to me since the `WithTest` is used extensively in our project. I would really appreciate it if you could help me understand why it is necessary. 
Could you please explain why it would be problematic to call the patch API without the `WithTest` function even if the annotation is already deleted? Thank you!

Reply:
I definitely agree that `WithTest` is used more than it should.

I'm not sure about this case in particular, but I'd ask myself: "Should the call to `RemoveDeschedulerAnnotationFromSourcePod()` fail if the pod's descheduler annotation is already removed?"

Reply:
pod resource is not controlled only by us. From my point of view, we should remove the annotations that we've added. In this case in particular, the annotation is a descheduler one.
It should not happen, but if an external component edits the annotation we should not drop it, because we are going in concurrency with something else, and we should care about it.
Consider that the test checks not only the existence but the value too.

Using the test/remove vs remove can have different outputs:
- with test: we test that the annotation is present and has a specific value
- with plain remove: we drop it without checking the value.
So if the annotation is there but with a different value:
- with test: we fail
- with remove: we succeed
Basically, I am trying to cover the case where the annotation has been modified by something/someone else.

Maybe it is better to set a better value(in place of `""`)...
Having said that, I can remove the test, but I would ask first: why shouldn't we check that?
Also, from your point of view, in which case we should use test? Thank you

Reply:
Two points regarding this:

1. I don't think we should consider users who modify launcher annotations. If someone does so, it is their responsibility.

2. For the question about why we shouldn't check this: the test operation in patch should be used when we want to ensure we move from one phase to another. For instance, if the annotation looks like this:

```
phase: <phase_number>
```
and we want to move from phase 2 to phase 3, we need to make sure that the current phase is phase 2 and it hasn't changed to phase 4 or phase 1, for instance.
In this case, the test operation would make sense. This is a valid scenario where the test operation should be used, in my opinion.

Another use case would be to acquire a lock. For instance, let's say we have a CRD called "lock" with two fields: one for the lock owner and the second is a boolean that indicates if the lock is taken. In that case, when someone wants to acquire the lock, they should make sure it is not locked anymore before they modify the owner field.

Reply:
I cannot see too many differences actually. If the phase is checked programmatically it is the same case as this.
Anyway, since the majority are for dropping it I followed your comment.
Thanks for the discussion